src/minimizer_mapper.cpp

```diff
@@ -895,20 +878,36 @@ pair<vector<Alignment>, vector< Alignment>> MinimizerMapper::map_paired(Alignmen
 
     // How many fragment clusters are at least as good as the one at each index
     vector<size_t> better_cluster_count (max_fragment_num+1); 
-    for (int j = fragment_cluster_indices_by_score.size() - 1 ; j >= 0 ; j--) {
-        size_t i = fragment_cluster_indices_by_score[j];
-        if (j == fragment_cluster_indices_by_score.size()-1) {
-            better_cluster_count[i] = j;
+
+    for (int rank = fragment_cluster_indices_by_score.size() - 1 ; rank >= 0 ; rank--) {
+        //Go through fragment clusters in descending score order and count how many equivalent or
+        //better clusters we found
+        size_t fragment_num = fragment_cluster_indices_by_score[rank];
+        if (rank == fragment_cluster_indices_by_score.size()-1) {
+            better_cluster_count[fragment_num] = rank+1;
         } else {
-            size_t i2 = fragment_cluster_indices_by_score[j+1];
-            if(cluster_coverage_by_fragment.first[i] + cluster_coverage_by_fragment.second[i] + cluster_score_by_fragment.first[i] + cluster_score_by_fragment.second[i] 
-                == cluster_coverage_by_fragment.first[i2] + cluster_coverage_by_fragment.second[i2] + cluster_score_by_fragment.first[i2] + cluster_score_by_fragment.second[i2]) {
-                better_cluster_count[i] = better_cluster_count[i2];
+            size_t prev_fragment_num = fragment_cluster_indices_by_score[rank+1];
+            if(cluster_coverage_by_fragment.first[fragment_num] +
+                cluster_coverage_by_fragment.second[fragment_num] +
+                cluster_score_by_fragment.first[fragment_num] +  
+                cluster_score_by_fragment.second[fragment_num] == 
+                cluster_coverage_by_fragment.first[prev_fragment_num] +
+                cluster_coverage_by_fragment.second[prev_fragment_num] +
+                cluster_score_by_fragment.first[prev_fragment_num] &&
+                cluster_score_by_fragment.second[prev_fragment_num]) {
+                //If this is the same as the last cluster, it has the same count
```

Comment:
I feel like there should be come named intermediate values computed here, instead of comparing two big computations. And can the `prev` result be saved between loops so there's only one copy of the computation?